src/Constants.h

```diff
@@ -0,0 +1,54 @@
+/*
+    Copyright 2016-2023 melonDS team
+
+    This file is part of melonDS.
+
+    melonDS is free software: you can redistribute it and/or modify it under
+    the terms of the GNU General Public License as published by the Free
+    Software Foundation, either version 3 of the License, or (at your option)
+    any later version.
+
+    melonDS is distributed in the hope that it will be useful, but WITHOUT ANY
+    WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+    FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with melonDS. If not, see http://www.gnu.org/licenses/.
+*/
+
+#pragma once
+
+#include "types.h"
+
+namespace melonDS
+{
+constexpr u32 MainRAMMaxSize = 0x1000000;
+constexpr u32 SharedWRAMSize = 0x8000;
+constexpr u32 ARM7WRAMSize = 0x10000;
+constexpr u32 NWRAMSize = 0x40000;
+constexpr u32 ARM9BIOSSize = 0x1000;
+constexpr u32 ARM7BIOSSize = 0x4000;
+
+// matching NDMA modes for DSi
+constexpr u32 NDMAModes[] =
```

Comment:
does this need to go into this file?

Otherwise I'd move it nearer to where it's used and rename this file to something more specifc than Constants.h

Reply:
What, `NDMAModes`? That particular declaration was pulled from a later commit in #1883, and the intent was to resolve an `#include` cycle that cropped up. Right now `NDS.cpp` includes `DSi.h`, but `DSi.cpp` includes `NDS.h`; if I don't move that declaration here, then `DSi.cpp` may not have access to `NDS`'s complete declarations. That's not an issue now, but it will be once I cherry-pick the next few commits.

Reply:
why did you move those out of the source file in the first place?

Reply:
To break `#include` cycles. You can reject this PR if you'd like, but the changes will probably have to be introduced later.

Reply:
https://github.com/search?q=repo%3AmelonDS-emu%2FmelonDS%20NDMAModes&type=code

can't you just leave the constant array in the cpp file where it came from? It doesn't need to be moved inside the class.

Reply:
I'll see what I can do. If I can't escape doing that, I'll document why.